Raygun4UWP/RaygunClient.cs

```diff
@@ -0,0 +1,455 @@
+      try
+      {
+        var tempFolder = ApplicationData.Current.TemporaryFolder;
+
+        var raygunFolder = await tempFolder.CreateFolderAsync(OFFLINE_DATA_FOLDER, CreationCollisionOption.OpenIfExists).AsTask().ConfigureAwait(false);
+
+        int number = 1;
+        while (true)
+        {
+          bool exists;
+
+          try
+          {
+            await raygunFolder.GetFileAsync($"RaygunCrashReport{number}.txt").AsTask().ConfigureAwait(false);
+            exists = true;
+          }
+          catch (FileNotFoundException)
+          {
+            exists = false;
+          }
+
+          if (!exists)
+          {
+            string nextFileName = $"RaygunCrashReport{number + 1}.txt";
+
+            StorageFile nextFile = null;
+            try
+            {
+              nextFile = await raygunFolder.GetFileAsync(nextFileName).AsTask().ConfigureAwait(false);
+
+              await nextFile.DeleteAsync().AsTask().ConfigureAwait(false);
+            }
+            catch (FileNotFoundException)
+            {
+            }
+
+            break;
+          }
+
+          number++;
+        }
+
+        if (number == 11)
+        {
+          try
+          {
+            StorageFile firstFile = await raygunFolder.GetFileAsync("RaygunCrashReport1.txt").AsTask().ConfigureAwait(false);
+            await firstFile.DeleteAsync().AsTask().ConfigureAwait(false);
+          }
+          catch (FileNotFoundException)
+          {
+          }
+        }
+
+        string fileName = $"RaygunCrashReport{number}.txt";
+        var file = await raygunFolder.CreateFileAsync(fileName).AsTask().ConfigureAwait(false);
+        await FileIO.WriteTextAsync(file, payload).AsTask().ConfigureAwait(false);
+
+        Debug.WriteLine($"Saved crash report: {OFFLINE_DATA_FOLDER}\\{fileName}");
+      }
+      catch (Exception ex)
+      {
+        Debug.WriteLine($"Error saving crash report to offline storage: {ex.Message}");
+      }
+    }
+
+    private RaygunCrashReport BuildCrashReport(Exception exception, IList<string> tags, IDictionary userCustomData, DateTime? currentTime)
+    {
+      string version = string.IsNullOrWhiteSpace(ApplicationVersion) ? GetPackageVersion() : ApplicationVersion;
+
+      var crashReport = RaygunCrashReportBuilder.New
+        .SetEnvironmentInfo()
+        .SetOccurredOn(currentTime)
+        .SetMachineName(new EasClientDeviceInformation().FriendlyName)
+        .SetErrorInfo(exception)
+        .SetClientInfo()
+        .SetVersion(version)
+        .SetTags(tags)
+        .SetCustomData(userCustomData)
+        .SetUserInfo(UserInfo ?? (!string.IsNullOrEmpty(UserIdentifier) ? new RaygunUserInfo(UserIdentifier) : null))
+        .Build();
+
+      return crashReport;
+    }
+
+    private string GetPackageVersion()
+    {
+      string version = null;
+
+      try
+      {
+        var v = Windows.ApplicationModel.Package.Current.Id.Version;
+        version = $"{v.Major}.{v.Minor}.{v.Build}.{v.Revision}";

```

Comment:
Could you just use `v.ToString()`?

Reply:
Nope, turns out that will just return "Windows.ApplicationModel.PackageVersion"